sdk/core/azure-core-test/README.md

```diff
@@ -1,108 +1,289 @@
+the `{ServiceName}AsyncClientTest` will be responsible for testing the asynchronous client. 
+The `{ServiceName}ClientTest` and the `{ServiceName}AsyncClientTest` both will extend the `{ServiceName}ClientTestBase` 
+which then extends the `TestProxyTestBase` class.
+The `{ServiceName}ClientTestBase` will be responsible for initializing the clients, preparing test data, registering
+sanitizers/matchers etc. (in this example we use Tables SDK for the sake of demonstration):
+
+```java
+public abstract class TableServiceClientTestBase extends TestProxyTestBase {
+    protected static final HttpClient DEFAULT_HTTP_CLIENT = HttpClient.createDefault();
+
+    protected HttpPipelinePolicy recordPolicy;
+    protected HttpClient playbackClient;
+
+    private TableServiceClientBuilder configureTestClientBuilder(TableServiceClientBuilder tableServiceClientBuilder) {
+        tableServiceClientBuilder
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS));
+
+        if (interceptorManager.isPlaybackMode()) {
+            playbackClient = interceptorManager.getPlaybackClient();
+            tableServiceClientBuilder.httpClient(buildAssertingClient(playbackClient));
+        } else {
+            tableServiceClientBuilder.httpClient(buildAssertingClient(DEFAULT_HTTP_CLIENT));
+            if (interceptorManager.isRecordMode()) {
+                tableServiceClientBuilder.addPolicy(recordPolicy);
+            }
+        }
+        if (!interceptorManager.isLiveMode()) {
+            TestUtils.addTestProxyTestSanitizersAndMatchers(interceptorManager);
+        }
+        return tableServiceClientBuilder;
     }
 }
 ```
 
-### Playback session records
+## Configure live or playback testing mode
 
-Playback test session records by creating a [RecordedData][RecordedData.java].
+"Live" tests refer to tests that make requests to actual Azure resources. "Playback" tests require a recording for each
+test; the test proxy will compare the requests/responses that would be made during each test with requests/responses in
+the recording.
 
-```java readme-sample-playbackSessionRecords
-/**
- * Sample code for using playback to test.
- */
-public class FooBar {
-    public void playbackNetworkCalls() {
-        RecordedData recordedData = new RecordedData();
+To run live tests, set the environment variable `AZURE_TEST_MODE` to `LIVE`.
+To run tests in playback, set `AZURE_TEST_MODE` to `PLAYBACK` or leave it unset.
 
-        // Add some network calls to be replayed by playbackClient
+## Run and record tests
 
-        // Creates a HTTP client that plays back responses in recordedData.
-        HttpClient playbackClient = new PlaybackClient(recordedData, null);
+Set the environment variable `AZURE_TEST_MODE` to `RECORD` to run your test(s) in record mode.
+
+After tests finish running, there should folder called `src/test/resources/session-records` in your package
```

Comment:
Is this still the case for a library that is already using the assets repo and we are just adding new tests to existing test classes?

Reply:
yes, once assets.json is present the new/updated files will be reflected in the respective `.assets` folder.